core/coreunix/metadata.go

```diff
@@ -14,7 +14,7 @@ import (
 func AddMetadataTo(n *core.IpfsNode, skey string, m *ft.Metadata) (string, error) {
 	ukey := key.B58KeyDecode(skey)
 
-	ctx, cancel := context.WithTimeout(context.TODO(), time.Minute)
```

Comment:
Notice that this newly prepared context *always* happens before `DAG.Get`. To clean this up is to put it inside `DAG.Get`